src/sickle.ts

```diff
@@ -163,28 +219,66 @@ class Annotator {
     // but it's needed to work around
     // https://github.com/Microsoft/TypeScript/issues/6982
     this.emit('\n/**\n');
-    let existingAnnotation = this.existingClosureAnnotation(fnDecl);
-    if (existingAnnotation) {
-      this.emit(' * ' + existingAnnotation + '\n');
+    let jsDoc = this.getJSDoc(fnDecl);
+    if (jsDoc) {
+      // Emit all the tags other than @param and @return.
+      for (let { tagName, text } of jsDoc.tags) {
+        if (tagName == 'param' || tagName == 'return') continue;
+        this.emit(' * ');
+        if (tagName) {
+          this.emit(`@${tagName}`);
+        }
+        if (text) {
+          this.emit(text);
+        }
+        this.emit('\n');
+      }
     }
     // Parameters.
     if (fnDecl.parameters.length) {
       for (let param of fnDecl.parameters) {
+        let paramDoc: string;
+        if (jsDoc) {
+          // Search for this parameter in the JSDoc @params.
+          let name = param.name.getText();
+          for (let { tagName, parameterName, text } of jsDoc.tags) {
+            if (tagName == 'param' && parameterName == name) {
+              paramDoc = text;
+              break;
+            }
+          }
+        }
         if (param.type) {
```

Comment:
Now you're dropping the docs if there's no type on the param...

What about formulating this as building up the JSDoc by taking existing JSDoc, then adding the type to the @param tag as needed (and adding empty docs for non-documented params), and then emitting the entire JSDoc?